crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -225,28 +225,76 @@ impl GitCommandProvider {
         })
     }
 
-    /// Open a repo, erroring if `path` is not a repo or is a subdirectory of a repo
-    pub fn open_with<P: AsRef<Path>>(
-        options: GitCommandOptions,
-        path: P,
-    ) -> Result<Self, GitCommandOpenError> {
-        let bare = {
+    fn is_bare(
+        options: &GitCommandOptions,
+        path: impl AsRef<Path>,
+    ) -> Result<bool, GitCommandOpenError> {
+        let mut command = options.new_command();
+        command
+            .args(["-C", path.as_ref().to_str().unwrap()])
+            .arg("rev-parse")
+            .arg("--is-bare-repository");
+
+        let out_str = GitCommandProvider::run_command(&mut command)?
+            .to_str()
+            .ok_or(GitCommandDiscoverError::GitDirEncoding)?
+            .to_string();
+
+        let bare = out_str
+            .trim()
+            .parse::<bool>()
+            .map_err(|_| GitCommandDiscoverError::UnexpectedOutput(out_str))?;
+        Ok(bare)
+    }
```

Comment:
suggeestion: rename `is_bare_repo`

It's kinda weird mixing instance methods with these "static" methods.
But if we do I'd prefer it has a name that sounds less like an "attribute".

```suggestion
    fn is_bare_repo(
        options: &GitCommandOptions,
        path: impl AsRef<Path>,
    ) -> Result<bool, GitCommandOpenError> {
        let mut command = options.new_command();
        command
            .args(["-C", path.as_ref().to_str().unwrap()])
            .arg("rev-parse")
            .arg("--is-bare-repository");

        let out_str = GitCommandProvider::run_command(&mut command)?
            .to_str()
            .ok_or(GitCommandDiscoverError::GitDirEncoding)?
            .to_string();

        let bare = out_str
            .trim()
            .parse::<bool>()
            .map_err(|_| GitCommandDiscoverError::UnexpectedOutput(out_str))?;
        Ok(bare)
    }
```

Reply:
suggestion: drop the `options` argument

I don't think we need to take options here (what options would affect rev-parse of a local repo, and how that is relevant?).

```suggestion
    fn is_bare_repo(
        path: impl AsRef<Path>,
    ) -> Result<bool, GitCommandOpenError> {
        let mut command = GitCommandOptions::default().new_command();
        command
            .args(["-C", path.as_ref().to_str().unwrap()])
            .arg("rev-parse")
            .arg("--is-bare-repository");

        let out_str = GitCommandProvider::run_command(&mut command)?
            .to_str()
            .ok_or(GitCommandDiscoverError::GitDirEncoding)?
            .to_string();

        let bare = out_str
            .trim()
            .parse::<bool>()
            .map_err(|_| GitCommandDiscoverError::UnexpectedOutput(out_str))?;
        Ok(bare)
    }
```